Admin/pages/AdminApproval.php

```diff
@@ -35,13 +35,15 @@ protected function initInternal()
 
 		$this->pending_ids = $this->getPendingIds();
 
-		if (count($this->pending_ids) == 0)
+		if (count($this->pending_ids) === 0) {
 			$this->relocate();
+		}
 
 		$this->id = $this->app->initVar('id');
 
-		if ($this->id === null)
+		if ($this->id === null) {
 			$this->id = $this->getNextId();
+		}
```

Comment:
Pretty sure how this works its that if you open the approval page without an id passed in, `$this->id === null` - but you can still look up the next id from the full list of pending ids.